### PR TITLE
fix: add processed types for all layers and layer configs

### PR DESCRIPTION
**Pull request by alittlebitweird**

### CODE REVIEW

File: src/constant/layerConfigs/audio.ts [hunk 1]
```diff
@@ -9,10 +9,18 @@ export type AudioOptions = {
   [AudioKey.volume]?: number
 }
 
+export type AudioOptionsProcessed = {
+  [AudioKey.volume]: number
+}
```
Comment (ptrkcsk) [inline src/constant/layerConfigs/audio.ts:14]: you could use [`Required`](https://www.typescriptlang.org/docs/handbook/utility-types.html#requiredtype) or [`SetRequired`](https://github.com/sindresorhus/type-fest/blob/main/source/set-required.d.ts)

**Review by ptrkcsk — APPROVED** [review]

File: src/constant/layerConfigs/audio.ts (hunk 1 above)
Comment (alittlebitweird) [reply src/constant/layerConfigs/audio.ts:14]: i started out that way but some attrs are required and some are not

Comment (alittlebitweird) [reply src/constant/layerConfigs/audio.ts:14]: ah, didn't know about `SetRequired`

Comment (alittlebitweird) [reply src/constant/layerConfigs/audio.ts:14]: redid them using SetRequired 🦄

**Pull request merged**